crates/telio-firewall/src/firewall.rs

```diff
@@ -610,6 +656,20 @@ impl StatefullFirewall {
     fn handle_inbound_icmp<'a, P: IpPacket<'a>>(&self, peer: &PublicKey, ip: &P) -> bool {
         let icmp_packet = unwrap_option_or_return!(P::Icmp::new(ip.payload()), false);
 
+        let key = IcmpConn {
+            remote_addr: ip.get_source().into(),
+            local_addr: ip.get_destination().into(),
+            identifier: 0,
```

Comment:
Also this should be filled in this function as well in case of echo response I guess

Reply:
I updated a function to create this key, that handles this and makes sure it's populated when it should be